evalml/tests/automl_tests/test_automl.py

```diff
@@ -824,25 +829,42 @@ def test_add_to_rankings(mock_fit, mock_score, dummy_binary_pipeline_class, X_y_
     X, y = X_y_binary
     mock_score.return_value = {'Log Loss Binary': 1.0}
 
-    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', max_iterations=1, allowed_pipelines=[dummy_binary_pipeline_class])
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', max_iterations=1,
+                          allowed_pipelines=[dummy_binary_pipeline_class])
     automl.search()
-    best_pipeline = automl.best_pipeline
-    assert best_pipeline is not None
-    mock_score.return_value = {'Log Loss Binary': 0.1234}
+    assert len(automl.rankings) == 1
+    assert len(automl.full_rankings) == 1
+    original_best_pipeline = automl.best_pipeline
+    assert original_best_pipeline is not None
 
+    mock_score.return_value = {'Log Loss Binary': 0.1234}
     test_pipeline = dummy_binary_pipeline_class(parameters={})
     automl.add_to_rankings(test_pipeline)
-    assert automl.best_pipeline != best_pipeline
-
+    assert automl.best_pipeline.name == test_pipeline.name
+    assert automl.best_pipeline.parameters == test_pipeline.parameters
+    assert automl.best_pipeline.component_graph == test_pipeline.component_graph
     assert len(automl.rankings) == 2
+    assert len(automl.full_rankings) == 2
     assert 0.1234 in automl.rankings['score'].values
 
+    mock_score.return_value = {'Log Loss Binary': 0.5678}
+    test_pipeline_2 = dummy_binary_pipeline_class(parameters={'Mock Classifier': {'a': 1.234}})
+    automl.add_to_rankings(test_pipeline_2)
+    assert automl.best_pipeline.name == test_pipeline.name
+    assert automl.best_pipeline.parameters == test_pipeline.parameters
+    assert automl.best_pipeline.component_graph == test_pipeline.component_graph
+    assert len(automl.rankings) == 2
+    assert len(automl.full_rankings) == 3
+    assert 0.5678 not in automl.rankings['score'].values
+    assert 0.5678 in automl.full_rankings['score'].values
```

Comment:
Why would this new score for `test_pipeline_2` not show up in full rankings? Seems like the full rankings got updated since the length did increase.

Reply:
@bchen1116 so in this case, we add a new pipeline with a worse log loss than the current best for the given model family. Under that scenario, I would expect `full_rankings` to have a row for the new pipeline, but `rankings` should remain unchanged, since the new pipeline doesn't beat the old for that family. Take a peek at the docstrings for `rankings` and `full_rankings` to see the intended behavior of those accessors.

Reply:
got it, somehow I read both assert lines to be `assert 0.5678 not in ...`. Thanks for clarifying!